### PR TITLE
FIX phpstan

### PR DESCRIPTION
**Pull request by inoveaconseil**

### CODE REVIEW

File: htdocs/core/class/commonobject.class.php [hunk 1]
```diff
@@ -755,6 +755,80 @@ abstract class CommonObject
 	// No constructor as it is an abstract class
 
 
+	/**
+	 * @var type
+	 */
+	public $type;
+
+	/**
+	 * @var status_buy
```
Comment (frederic34) [inline htdocs/core/class/commonobject.class.php:764]: int

File: htdocs/core/class/commonobject.class.php [hunk 2]
```diff
@@ -755,6 +755,80 @@ abstract class CommonObject
...
+	 * @var string tpl
+	 */public $tpl;
+
+	/**
+	 * @var int socid
+	 */public $socid;
+
+	/**
+	 * @var status_buy
+	 */public $status_buy;
```
Comment (frederic34) [inline htdocs/core/class/commonobject.class.php:777]: duplicate

File: htdocs/core/class/commonobject.class.php [hunk 3]
```diff
@@ -755,6 +755,80 @@ abstract class CommonObject
...
+	 * @var int socid
+	 */
+	public $socid;
+
+
+
+	/**
+	 * @var float imgHeight
+	 */
+	public $imgHeight;
```
Comment (eldy) [inline htdocs/core/class/commonobject.class.php:783]: Why do we need this property as common ?

File: htdocs/core/class/commonobject.class.php [hunk 4]
```diff
@@ -755,6 +755,80 @@ abstract class CommonObject
...
+
+	/**
+	 * @var float imgWidth
+	 */
+	public $imgWidth;
+
+	/**
+	 * @var int nbphoto
+	 */
+	public $nbphoto;
```
Comment (eldy) [inline htdocs/core/class/commonobject.class.php:793]: Why do we need this property as common ?

File: htdocs/core/class/commonobject.class.php [hunk 5]
```diff
@@ -755,6 +755,80 @@ abstract class CommonObject
...
+
+	/**
+	 * @var string description
+	 */
+	public $description;
+
+	/**
+	 * @var int fk_user_creat
+	 */
+	public $fk_user_creat;
```
Comment (eldy) [inline htdocs/core/class/commonobject.class.php:813]: Deprecated, field user_author_id must be used

File: htdocs/core/class/commonobject.class.php [hunk 6]
```diff
@@ -755,6 +755,80 @@ abstract class CommonObject
...
+
+	/**
+	 * @var string label
+	 */
+	public $label;
+
+	/**
+	 * @var float qty
+	 */
+	public $qty;
```
Comment (eldy) [inline htdocs/core/class/commonobject.class.php:828]: What is a qty on an object. Qty is on objectline only.

**Comment by inoveaconseil** [comment]
PHPstan return this error:
tdocs/core/class/commonobject.class.php	9107	Access to an undefined property CommonObject::$imgHeight.
htdocs/core/class/commonobject.class.php	9107	Access to an undefined property CommonObject::$imgWidth.
htdocs/core/class/commonobject.class.php	9173	Access to an undefined property CommonObject::$nbphoto

 So we can search the class who used them, but why phpstan reported them.

**Comment by eldy** [comment]
> PHPstan return this error:
> tdocs/core/class/commonobject.class.php	9107	Access to an undefined property CommonObject::$imgHeight.
> htdocs/core/class/commonobject.class.php	9107	Access to an undefined property CommonObject::$imgWidth.
> htdocs/core/class/commonobject.class.php	9173	Access to an undefined property CommonObject::$nbphoto
> 
>  So we can search the class who used them, but why phpstan reported them.

Pb was probably in classes that was using this properties. It seems it has been fixed by other since the mentionned lines has disappeared from apstats report.

**Pull request closed without merging**